README.md

```diff
@@ -45,157 +53,294 @@ console.log(result);
-- validateAge
-- validateEmail
-- validateReason
+## Problem
+As part of an application that serves ~~kittehs~~ users wishing to apply to be an astronaut, you're running a server which accepts POST requests of a forms content to the path /submit. You need to validate the form contents before providing an appropriate response to the ~~kitteh~~ user, so you'll need write several functions inside of the handler dealing with requests to /submit, and write tests for these functions.
 
-Working in pairs, clone this repo to get started with the basic file structure and dependencies. You should utilise the **Returning errors** and/or **Error first callback** methods described above,
-and be sure to cover as many cases as possible in your tests.
+## Specification
+##### `POST /submit`
+This endpoint accepts the following parameters in the body of the request (also known as the payload):
 
-The functions to build are located in the *modules* directory and the tests are located in the *tests* directory.
+* `filename`: String, filename with which the application will be saved. 30 characters or fewer. Beware of [directory traversal!](https://www.owasp.org/index.php/Path_Traversal)
+* `contents`: Object, with the following attributes...
+  * `name`: String, name of the applicant, 30 characters or fewer.
+  * `age`: Integer, between 2 and 20.
+  * `body`: String, body of the application text, 10000 characters or fewer.
 
-### (Bonus exercise if you have time)
+Once the handler has validated the parameters of the request it should call the provided `createRecord` function, which has the following signature:
+```js
+createRecord(filename, contents, callback)
+```
+* `filename`: Filename to save to, same definition as above.
+* `contents`: Data structure defining the book contents with the same definition as above.
+* `callback`: A function with the signature `callback(error, message)`, where `error` is either `null` or a potential `Error` object, `message` contains the response to the client.
 
-Build a *router* for your server to handle POST requests to **/submit**. This example should help with dealing with post requests;
-```javascript
-const qs = require('querystring');
+**Note**: the `createRecord` function performs no error checking and assumes all its inputs are correct. Incorrect inputs will have unpredictable consequences.
 
-const router = (request, response) => {
-    if (request.method == 'POST') {
-        let body = '';
+## Other Requirements
+* The server should not crash because of a client request.
+* If the client supplies bad data, the server should respond with a `400` (Bad request) status code and a helpful message indicating the problem.
 
-        request.on('data', (data) => {
-            body += data;
-        });
+## Steps
+1. Decide how your validation code will indicate that there is a problem with the payload. You are free to use any of the three approaches outlined above (or use another approach if you feel it is appropriate).
```

Comment:
Nothing in principle, although that solution doesn't satisfy the following requirement above:
> If the client supplies bad data, the server should respond with a `400` (Bad request) status code and a helpful message indicating the problem.